### DISCUSSION TITLE
How to Manage "To confirm your identity, sign in with the same account you used originally."

### DISCUSSION THREAD

**Discussion by HenriTeinturier** [nextauthjs/next-auth#10631] [Help]
Hello.
I have 2 providers (gihub and google). I understant that it's not secure to accept 2 provider with same email.

I would like to manage this error and display a page for explain users to login with the good provider.
/api/auth/signin?error=OAuthAccountNotLinked
I'm with nextjs 14 and auth.js 5.

should I force allowDangerousEmailAccountLinking? and how to?
I prefer create a page.tsx or error.tsx to explain to user. And the user can then retry a connection using the correct provider.

This is my auth function
```javascript
export const {
  handlers,
  signIn,
  signOut,
  auth: baseAuth,
} = NextAuth({
  adapter: PrismaAdapter(prisma),

  providers: [GitHub, Google],
  secret: process.env.AUTH_SECRET,
    },
});
  
```

**Comment by HenriTeinturier** [comment]
Finally i do this
```javascript
export const {
  handlers,
  signIn,
  signOut,
  auth: baseAuth,
} = NextAuth({
  adapter: PrismaAdapter(prisma),

  providers: [
    GitHub({ allowDangerousEmailAccountLinking: true }),
    Google({ allowDangerousEmailAccountLinking: true }),
  ],
  ```

**Reply by Capta1nRaj** [reply]
GOAT answer fr.

**Comment by Larusyang** [comment]
great answer ,just solve this problem

**Reply by vishal590** [reply]
I am using only discord oauth. Then why I Am getting thus error

**Reply by Capta1nRaj** [reply]
> I am using only discord oauth. Then why I Am getting thus error

Tried this?

https://github.com/nextauthjs/next-auth/discussions/10631#discussioncomment-9156852

**Comment by ArmandoEnrique1010K** [comment]
Tranks you
